include/ensmallen_bits/problems/nsga2_test_sch_function.hpp

```diff
@@ -0,0 +1,61 @@
+/**
+ * @file nsga2_test_sch_function.hpp
+ * @author Sayan Goswami
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef ENSMALLEN_PROBLEMS_NSGA2_TEST_SCH_FUNCTION_HPP
+#define ENSMALLEN_PROBLEMS_NSGA2_TEST_SCH_FUNCTION_HPP
+
+namespace ens {
+namespace test {
+
+template<typename MatType = arma::mat>
+class NSGA2TestFuncSCH {
```

Comment:
I think we should go for a name like SchafferStudyFunction or SchaffarMultiObjFunction, since that would be consistent with the rest of the library, plus the function could apply to more than NSGA2.

Reply:
Yes, I was thinking of building a generic ABC for multi-objective functions and building up SCH on top of it.